mlrun/datastore/base.py

```diff
@@ -388,6 +615,28 @@ def _is_dd(df_module):
         except ImportError:
             return False
 
+    @staticmethod
+    def verify_path_partition_level(path: str, partitions: list[str]) -> bool:
+        if not partitions:
+            return False
+
+        path_parts = urlparse(path).path.strip("/").split("/")
+        path_parts = [part.split("=")[0] for part in path_parts if "=" in part]
+        if "hour" in partitions:
+            hour_index = partitions.index("hour")
+        else:
+            return False
+        for i, part in enumerate(partitions):
+            if (
+                part in path_parts
+                or part in ["year", "month", "day", "hour"]
+                or i > hour_index
+            ):
+                continue
+            else:
+                return False
```

Comment:
Can be simplified a bit:
```suggestion
            if not (
                part in path_parts
                or part in ["year", "month", "day", "hour"]
                or i > hour_index
            ):
                return False
```